full-service/src/db/account.rs

```diff
@@ -276,23 +276,18 @@ impl AccountModel for Account {
     fn get_by_txo_id(
         txo_id_hex: &str,
         conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
-    ) -> Result<Account, WalletDbError> {
+    ) -> Result<Vec<Account>, WalletDbError> {
         use crate::db::schema::account_txo_statuses::dsl::account_txo_statuses;
 
         match account_txo_statuses
             .select(crate::db::schema::account_txo_statuses::all_columns)
             .filter(crate::db::schema::account_txo_statuses::txo_id_hex.eq(txo_id_hex))
             .load::<AccountTxoStatus>(conn)
         {
-            Ok(a) => {
-                if a.len() > 1 {
-                    return Err(WalletDbError::MultipleStatusesForTxo);
-                }
-                Ok(Account::get(
-                    &AccountID(a[0].account_id_hex.to_string()),
-                    conn,
-                )?)
-            }
+            Ok(accounts) => Ok(accounts
```

Comment:
When we have multiple accounts in the same wallet, then when getting a Txo, we need to return all accounts which may be associated with that Txo.